starknet_py/tests/e2e/block_test.py

```diff
@@ -1,10 +1,13 @@
 import pytest
 
+from starknet_py.compile.compiler import Compiler
 from starknet_py.contract import Contract
+from starknet_py.tests.e2e.fixtures.constants import MAX_FEE
 
 
 @pytest.mark.asyncio
-async def test_pending_block(gateway_account_client):
+async def test_pending_block(new_gateway_account_client):
```

Comment:
Could we merge and parametrize / extract common parts of test_pending_block and test_latest_block?